collectors/bzimport/convertors.py

```diff
@@ -301,6 +305,15 @@ def clean_meta(self):
         for meta in to_delete:
             meta.delete()
 
+    def clean_acknowledgments(self):
+        """clean obsoleted flaw acknowledgments"""
+        old_acknowledgments = set(self.flaw.acknowledgments.all())
+        new_acknowledgments = set(self.acknowledgments)
+
+        to_delete = list(old_acknowledgments - new_acknowledgments)
```

Comment:
I am not sure this would catch the case when an acknowledgment changes the `from_upstream` boolean. Not saying that it will not be rewritten later just not sure so rather check it.

Reply:
Right, that has not been tested and I enhanced `test_acknowledgment_removed` to test it.